cal/cal/static/js/app.js

```diff
@@ -1,111 +1,124 @@
-var analyticsApp = angular.module('analyticsApp', ['nvd3', 'ui.calendar']);
+/*jslint devel: true */
+/*jslint browser: true */
+/*jslint jquery: true */
+/*global d3 */
+/*global getCookie */
+/*global moment */
 
-analyticsApp.controller('LoggedInCtrl', function LoggedInController($scope) {
+var analyticsApp = window.angular.module('analyticsApp', ['analyticsApp.services', 'nvd3', 'ui.calendar']);
+
+analyticsApp.controller('LoggedInCtrl', function LoggedInController() {
 });
 
-function TagsCtrl($scope, $http) {
-  var tagUrl = '/v1/tags';
-  $scope.tags = [];
+function TagListCtrl($scope, $http, CalendarRangeService, TagService) {
 
-  // add all the tags
-  $http({method: 'GET', url: tagUrl + '.json' }).
-  success(function successCallback(data) {
-    for (var i = 0; i < data.results.length; i++) {
-      var tag = data.results[i];
-      $scope.tags.push({
-        id: tag.id,
-        label: tag.label,
-        keywords: tag.keywords,
-        hours: tag.hours
-      });
+  var _this = this;
+
+  this.tags = [];
+
+  $scope.$on('calendarRange:updated', function(event, data) {
+    /* jshint unused:vars */
+    var rangeData = CalendarRangeService.getRange();
+    var timeRange = rangeData.timeRange;
+    if (!_this.isCumulative) {
+      _this.getTags('cumulative', null, null)
+        .then(function(tags) {
+          _this.tags = tags;
+          _this.timeRange = timeRange;
+        });
     }
   });
 
+  // Initialization
+  this.initialize = function() {
+
+    var tagsPromise;
+
+    if (this.isCumulative) {
+      tagsPromise = TagService.getTags('cumulative', null, null);
+    } else {
+      var initialTimeRange = CalendarRangeService.getRange();
+      tagsPromise = TagService.getTags(initialTimeRange.timeRange, initialTimeRange.start, initialTimeRange.end);
+    }
+    tagsPromise.then(function(tags) {
+      _this.tags = tags;
+    });
+  }.bind(this);
+  this.initialize();
```

Comment:
It gets the initial tags for the component, no matter whether it is daily/weekly/monthly/cumulative